get_openstack_info_open_port/get_openstack_info_open_port/get_openstack_info_open_port.py

```diff
@@ -0,0 +1,190 @@
+        return cred
+
+    def _os_auth(self):
+        self.keystone_session = session.Session(
+            auth=v3.Password(**self._get_credentials()))
+        self.keystone_v3 = keystoneclient_v3.Client(
+            session=self.keystone_session)
+        self.nova = nova.Client("2.1", session=self.keystone_session)
+        self.neutron = neutron.Client("2.0", session=self.keystone_session)
+        
+    def _check_openstack(self):
+        ''' Check that OpenStack variables are loaded '''
+        if 'OS_USERNAME' not in os.environ:
+            self._log.error("You don't seem to have loaded a source file with credentials to OpenStack. Please do so, and try again.")
+            sys.exit(1)
+
+
+    def _get_openstack_instances(self):
+        self.server_list = self.nova.servers.list(detailed=True, search_opts={"all_tenants": True})
+        self._log.debug(f"Obtained {len(self.server_list)} servers.")
+        
+        
+    def _locate_affected_instance(self):
+        self.affected_instance = None
+        for server in self.server_list:
+            server_info = server.to_dict()
+            for project in server_info['addresses'].keys():
+                for address in server_info['addresses'][project]:
+                    if address['addr'] == self.ip:
+                        self._log.debug(f"Located instance '{server_info['name']}' with ID '{server_info['id']}' and IP '{address['addr']}' in project '{project}'")
+                        self.affected_instance = server_info
+                        #print(json.dumps(self.affected_instance, indent=2))
+        if self.affected_instance is None:
+            self._log.error(f"Not found any instance with IP '{self.ip}'")
+            sys.ext(2)
+            
+    def _check_affected_security_groups(self):
+        all_security_groups_in_project = self.neutron.list_security_groups(tenant_id=self.affected_instance['tenant_id'])['security_groups']
+        for security_group in self.affected_instance['security_groups']:
+            self._log.debug(f"Looking for security group '{security_group['name']}'...")
+            for existing_sec_group in all_security_groups_in_project:
+                if existing_sec_group['name'] == security_group['name']:
+                    for rule in existing_sec_group['security_group_rules']:
+                        if rule.get('direction', '') == 'ingress':
+                            if rule.get('remote_ip_prefix', '') == '0.0.0.0/0':
+                                port_range_max = rule.get('port_range_max', 0)
+                                if port_range_max is not None and port_range_max <= self.port:
+                                    port_range_mi = rule.get('port_range_min', 0)
+                                    if port_range_mi is not None and port_range_mi >= self.port:
+                                        self._log.info(f"Rule '{rule['id']}' in security group '{security_group['name']}' allows incoming traffic.")
```

Comment:
For this. Openstack has a bug that an ip 1.2.3.4/0 will get translated as 0.0.0.0/0 in iptables.

It might be good to check for this case as well.

Reply:
I added a rule with that CIDR and my code didn't translate it to 0.0.0.0/0, it might be a CLI bug but not a Python module one?

Reply:
Changed to check for the netmask instead